### PR TITLE
Move platform docs to guides (they were never howtos)

### PR DESCRIPTION
**Pull request by mit-mit**

### CODE REVIEW

File: _data/sidebars/home_sidebar.yml [hunk 1]
```diff
@@ -74,6 +74,14 @@ entries:
       url: /widgets/
       output: web
 
+    - title: Plugins & platform-specific code
```
Comment (sethladd) [inline _data/sidebars/home_sidebar.yml]: Did we come up with something to replace "platform-specific"? That's too easy to misinterpret as "cupertino vs material design".

Strawman: "Plugins for SDKs and APIs"

File: _data/sidebars/home_sidebar.yml [hunk 2]
```diff
@@ -74,6 +74,14 @@ entries:
...
+      url: /platform-plugins/
+      output: web
+
+    - title: Custom platform-specific code with platform channels
```
Comment (sethladd) [inline _data/sidebars/home_sidebar.yml]: What's our convention? "PlatformChannel" or "platform channel" ?

File: _data/sidebars/home_sidebar.yml (hunk 1 above)
Comment (mit-mit) [inline _data/sidebars/home_sidebar.yml]: No we didn't. "Platform-specific code" is what we started using when we deemed "Native code" too confusing. I'm worried about doing another "indirection".

File: _data/sidebars/home_sidebar.yml (hunk 2 above)
Comment (mit-mit) [inline _data/sidebars/home_sidebar.yml]: Platform channel is the overarching concept, so we use that generally. There is actually no type called `PlatformChannel` so we don't use that (there are several specific ones, e.g. `PlatformMethodChannel`)

**Comment by sethladd** [comment]
lgtm

**Pull request merged**